### PR TITLE
Updated Inference Server for GPT Finetuner

### PR DESCRIPTION
**Pull request by harubaru**
The previous GPT inference server used in the finetuning example was out of date to the point that it cannot run inference on Pythia or GPT-NeoX models. The changes in this PR also incorporates an inference server with the finetuner image itself for simplicity.

Sidenote, #198 was made out of accident due to me being a little sleep deprived :sweat_smile:

### CODE REVIEW

File: finetuner-workflow/finetuner/requirements.txt [hunk 1]
```diff
@@ -8,3 +8,4 @@ accelerate~=0.17.1
 tensorizer==1.0.1
 einops
 flash-attn
+kserve
```
Comment (NavarrePratt) [inline finetuner-workflow/finetuner/requirements.txt:11]: Few things:

1. We should include some sort of versioning for all the packages. That way everytime a customer builds an image going through the docs, we know it'll be consistent (as much as we can control at least)

2. The pip install fails both locally and in the docker build. Were you able to build the image and test it in the workflow? In the workflow it's still using the goose finetuner image.

3. Should we be bumping tensorizer to the newest version, 1.1.0?

File: finetuner-workflow/finetuner/Dockerfile [hunk 2]
```diff
@@ -12,5 +12,6 @@ RUN pip3 install --no-cache-dir -r requirements.txt
 COPY ds_config.json .
 COPY finetuner.py .
 COPY evaluator.py .
+COPY inference.py .
```
Comment (NavarrePratt) [inline finetuner-workflow/finetuner/Dockerfile:15]: I'm fine with doing this to get everything working ASAP, but we should be making the inference and finetuner images separate.

We at least don't need `deepspeed` in the inference image, and there's probably some other unused dependencies as well.

This is also a general huggingface-transformers-kserve inference script. Should this live somewhere separate like `ml-images`? I think we've been in need of putting all our the nested images in kubernetes-cloud someone else more visible.

**Comment by harubaru** [comment]
For reference, this PR is to be added on top of https://github.com/coreweave/kubernetes-cloud/pull/128

**Review by wbrown — APPROVED** [review]
@harubaru I'll approve and merge this, but the workflow should really be passing in the arguments that we have for sampling.

Also, we should be sharing sampling code where possible with `evaluate.py`.

**Pull request merged**